drivers/flash/spi_flash_at45.c

```diff
@@ -340,6 +347,10 @@ static int spi_flash_at45_write(const struct device *dev, off_t offset,
 
 	acquire(dev);
 
+	if (cfg->wp_gpio) {
+		gpio_pin_set(cfg->wp_gpio, cfg->wp_pin, 1);
```

Comment:
I guess the WP pin is supposed to be deasserted before a write or erase operation. The above does the opposite.

Reply:
Chip write-protected when WP pin is Low state. Setting 1 to WP enables writing.

Reply:
@anangl Ah, sure, you are right. "protection" means inverted logic, so better to use inverted value and "active-low" definition.